File: Data Structures & Algorithms/house-robber/submission-0.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 1) {
            return nums[0];
        }
        
        int h0 = nums[0];
        int h1 = Math.max(nums[0], nums[1]);
        int h2 = h1;
        for (int j = 2; j < nums.length; j++) {
            h2 = Math.max(nums[j] + h0, h1);
            h0 = h1;
            h1 = h2;
        }
        return h2;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        def houseRobber(nums):
            rob0, rob1 = 0,0
            for n in nums:
                temp = max(n + rob0, rob1)
                rob0 = rob1
                rob1 = temp
            return rob1
        robbing0 = nums[0] + houseRobber(nums[2:-1])
        notRobbing0 = houseRobber(nums[1:])
        return max(robbing0, notRobbing0)